pkg/apis/resource/v1alpha3/defaults.go

```diff
@@ -26,6 +26,19 @@ func addDefaultingFuncs(scheme *runtime.Scheme) error {
 }
 
 func SetDefaults_DeviceRequest(obj *resourceapi.DeviceRequest) {
+	if len(obj.FirstAvailable) > 0 {
```

Comment:
I suppose it is the same result, but this feels like it should be `if obj.DeviceClassName != ""` -- that's really the trigger for the one-of logic, isn't it?